integration_test.go

```diff
@@ -1540,25 +1539,46 @@ func TestDeadlockConsumerIssue48(t *testing.T) {
 
 // https://github.com/streadway/amqp/issues/46
 func TestRepeatedChannelExceptionWithPublishAndMaxProcsIssue46(t *testing.T) {
-	if strings.Compare(os.Getenv("CI"), "true") == 0 {
-		t.Skip("FLAKY - https://github.com/rabbitmq/amqp091-go/issues/77")
-	}
-	conn := integrationConnection(t, "issue46")
-	if conn != nil {
-		t.Cleanup(func() { conn.Close() })
+	var conn *Connection = nil
 
-		for i := 0; i < 100; i++ {
-			ch, err := conn.Channel()
-			if err != nil {
-				t.Fatalf("expected error only on publish, got error on channel.open: %v", err)
+	t.Cleanup(func() {
+		if conn != nil {
+			conn.Close()
+		}
+	})
+
+	for i := 0; i < 100; i++ {
+		if conn == nil || conn.IsClosed() {
+			conn = integrationConnection(t, "issue46")
+			if conn == nil {
+				t.Fatal("conn is nil")
+			}
```

Comment:
How could this condition happen?
conn is never changed after it is opened, and integrationConnection() always returns a non-nil conn